concordium-smart-contract-testing/src/lib.rs

```diff
@@ -0,0 +1,2745 @@
+                            println!("Querying account balance of {}", address);
+                            // When querying an account, the amounts from any `invoke_transfer`s
+                            // should be included. That is handled by
+                            // the `chain` struct already. transaction.
+                            // However, that is hand
+                            let response = match self.chain.account_balance(address) {
+                                Some(acc_bal) => {
+                                    // If you query the invoker account, it should also
+                                    // take into account the send-amount and the amount reserved for
+                                    // the reserved max energy. The former is handled in
+                                    // `contract_update_aux`, but the latter is represented in
+                                    // `self.invoker_amount_reserved_for_nrg`.
+                                    let acc_bal = if address == self.invoker {
+                                        acc_bal - self.invoker_amount_reserved_for_nrg
+                                    } else {
+                                        acc_bal
+                                    };
+
+                                    // TODO: Do we need non-zero staked and shielded balances?
+                                    let balances =
+                                        to_bytes(&(acc_bal, Amount::zero(), Amount::zero()));
+                                    InvokeResponse::Success {
+                                        new_balance: self
+                                            .chain
+                                            .get_instance(self.address)?
+                                            .self_balance,
+                                        data:        Some(balances),
+                                    }
+                                }
+                                None => InvokeResponse::Failure {
+                                    kind: v1::InvokeFailure::NonExistentAccount,
+                                },
+                            };
+
+                            let energy_after_invoke = remaining_energy - QUERY_ACCOUNT_BALANCE_COST;
+
+                            let resume_res = v1::resume_receive(
+                                config,
+                                response,
+                                InterpreterEnergy::from(energy_after_invoke),
+                                &mut self.mutable_state,
+                                false, // State never changes on queries.
+                                self.loader,
+                            );
+
+                            self.process(resume_res)
+                        }
+                        v1::Interrupt::QueryContractBalance { address } => {
+                            println!("Querying contract balance of {}", address);
```

Comment:
I assume all println! will be removed? 

This is meant to be used as a library, and libraries should not be printing stuff without any control.

If we want to we could set the chain up with a logger and use that here consistenly. That would be helpful for debugging. But it should be controllable.

Reply:
Yes, it is for debugging during development of this.
Logging would be nice, yes.